### PR TITLE
More updates to the CHANGES file

### LINKED ISSUE #14476: Missing CHANGES entry for text output differences
There are cosmetic differences in the text output from some functions and command line apps as identified in #14275 where the 1.1.1 tests were run against the 3.0 libraries.

We should have a CHANGES entry that describes this difference.

test_crl was failing because of a cosmetic difference in textual output between
1.1.1 and 3.0. One of the tests prints a crl using "-text" and compares it
against a reference output. In 3.0 printing a CRL adds an additional label for
the signature value, as well as a difference in indentation for the signature
value:

```` diff
diff --git a/test/certs/cyrillic_crl.utf8 b/test/certs/cyrillic_crl.utf8
index e55c549d4c..c36a3fd0f0 100644
--- a/test/certs/cyrillic_crl.utf8
+++ b/test/certs/cyrillic_crl.utf8
@@ -9,21 +9,22 @@ Certificate Revocation List (CRL):
                 1
 No Revoked Certificates.
     Signature Algorithm: sha256WithRSAEncryption
-         85:e5:e5:fe:d4:13:3f:07:1a:07:53:6f:f7:a5:01:c9:80:f4:
-         8a:7a:f3:74:fc:af:dd:6a:21:47:88:99:7b:29:bf:46:0b:02:
-         98:d7:80:75:46:f6:cd:da:b5:0f:ff:9f:0c:b7:e6:aa:8e:f6:
-         ae:7f:5c:81:ce:56:89:41:e2:4a:65:cb:02:98:6f:69:e9:3a:
-         f5:cb:40:49:5e:1a:ea:e6:40:b4:48:bc:8d:0e:c7:c6:51:37:
-         ee:c6:7c:26:a3:e7:25:1d:74:54:fa:02:ae:93:e8:74:5a:42:
-         4b:d3:6d:99:61:b2:77:f3:0e:7c:f2:4e:0d:4f:70:96:54:77:
-         84:db:71:03:0b:6e:b8:a7:de:36:9a:50:c4:ed:e9:fa:33:22:
-         f5:e7:63:de:2d:df:58:ad:68:aa:e6:23:88:3a:b2:1b:40:b1:
-         2b:ab:41:23:c3:1b:c7:1b:db:0c:89:81:54:6b:a5:6b:c2:64:
-         5f:db:a9:f2:67:bb:bd:44:a2:43:3f:ea:36:76:5b:70:76:20:
-         5a:49:70:b7:42:dd:e9:67:c0:48:7e:ff:e5:f3:59:70:a7:c0:
-         eb:eb:74:b0:08:82:36:a4:84:69:97:22:02:75:0a:a6:5f:f6:
-         be:d6:af:24:4e:15:c8:3f:62:b1:f9:7e:3c:83:b5:44:da:8d:
-         97:e1:c7:f2
+    Signature Value:
+        85:e5:e5:fe:d4:13:3f:07:1a:07:53:6f:f7:a5:01:c9:80:f4:
+        8a:7a:f3:74:fc:af:dd:6a:21:47:88:99:7b:29:bf:46:0b:02:
+        98:d7:80:75:46:f6:cd:da:b5:0f:ff:9f:0c:b7:e6:aa:8e:f6:
+        ae:7f:5c:81:ce:56:89:41:e2:4a:65:cb:02:98:6f:69:e9:3a:
+        f5:cb:40:49:5e:1a:ea:e6:40:b4:48:bc:8d:0e:c7:c6:51:37:
+        ee:c6:7c:26:a3:e7:25:1d:74:54:fa:02:ae:93:e8:74:5a:42:
+        4b:d3:6d:99:61:b2:77:f3:0e:7c:f2:4e:0d:4f:70:96:54:77:
+        84:db:71:03:0b:6e:b8:a7:de:36:9a:50:c4:ed:e9:fa:33:22:
+        f5:e7:63:de:2d:df:58:ad:68:aa:e6:23:88:3a:b2:1b:40:b1:
+        2b:ab:41:23:c3:1b:c7:1b:db:0c:89:81:54:6b:a5:6b:c2:64:
+        5f:db:a9:f2:67:bb:bd:44:a2:43:3f:ea:36:76:5b:70:76:20:
+        5a:49:70:b7:42:dd:e9:67:c0:48:7e:ff:e5:f3:59:70:a7:c0:
+        eb:eb:74:b0:08:82:36:a4:84:69:97:22:02:75:0a:a6:5f:f6:
+        be:d6:af:24:4e:15:c8:3f:62:b1:f9:7e:3c:83:b5:44:da:8d:
+        97:e1:c7:f2
 -----BEGIN X509 CRL-----
 MIIB6DCB0QIBATANBgkqhkiG9w0BAQsFADCBjjELMAkGA1UEBhMCUlUxFTATBgNV
 BAgMDNCc0L7RgdC60LLQsDELMAkGA1UECgwC0K8xCzAJBgNVBAsMAtCvMSowKAYD
 ````

test_x509 was failing in a simiar way for 2 of its tests. The cosmetic changes
were very different. In addition we no longer print the redundant label "keyid:" 
in front of the hex when printing the Authority key identifier. This makes the
output consistent with the way the Subject key identifier is printed.

### LINKED ISSUE #14477: Missing CHANGES entry for EVP_PKEY_public_check() and EVP_PKEY_param_check()
EVP_PKEY_public_check() and EVP_PKEY_param_check() now work for RSA keys
Previously calling these functions with an RSA key would return -2 (unsupported).
This is now supported in 3.0 and consequently they return 0 or 1. The
evp_extra_test was failing because of this.

There should be a CHANGES entry for this.

### LINKED ISSUE #14482: ECX keys always fail EVP_PKEY_param_check()
ECX keys don't have parameters. Neither do RSA keys. RSA keys always pass EVP_PKEY_param_check() but ECX keys always fail.

I think, where it is valid for an EVP_PKEY to have no parameters, then EVP_PKEY_param_check() should always pass in this situation.

### PR DESCRIPTION
**Pull request by mattcaswell**
Add CHANGES entries for cosmetic text output from various functions and the apps, as well as the newly implemented EVP_PKEY_param_check() and EVP_PKEY_public_check() functions for more key types.

As part of this change we also fix EVP_PKEY_param_check() for ECX keys.

Fixes #14476
Fixes #14482 
Fixes #14477

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -22,6 +22,20 @@ OpenSSL 3.0
 -----------
 
 ### Changes between 1.1.1 and 3.0 [xx XXX xxxx]
+
+ * The EVP_PKEY_public_check() and EVP_PKEY_param_check() functions now work for
+   more key types include RSA, DSA, ED25519, X25519, ED448 and X448. Previously
+   (in 1.1.1) they would always return -2. For key types that do not have
+   parameters then EVP_PKEY_param_check() will still return 1.
+
+ * The output from numerous "printing" functions such X509_signature_print(),
```
Comment (slontis) [inline CHANGES.md:31]: such as

File: CHANGES.md [hunk 2]
```diff
@@ -22,6 +22,20 @@ OpenSSL 3.0
...
+   X509_print_ex(), X509_CRL_print_ex(), and other similar functions has been
```
Comment (slontis) [inline CHANGES.md:32]: has=have

File: CHANGES.md [hunk 3]
```diff
@@ -22,6 +22,20 @@ OpenSSL 3.0
...
+   amended such that there may be cosmetic differences between the output
+   observed in 1.1.1 and 3.0. This also applies to the "-text" output from the
+   x509 and crl applications.
+
+   *David von Oheimb*
```
Comment (slontis) [inline CHANGES.md:37]: Author??

File: CHANGES.md (hunk 2 above)
Comment (mattcaswell) [reply CHANGES.md:32]: Hmm, no? You was say "The output has been", not "The output have been".

File: CHANGES.md (hunk 3 above)
Comment (mattcaswell) [reply CHANGES.md:37]: @DDvO made the cosmetic changes that this entry is referring to, so I credited him in the CHANGES entry.

**Comment by mattcaswell** [comment]
All feedback either addressed or responded to. Please take another look.

**Review by slontis — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**